lib/ansible/modules/windows/win_disk_management.ps1

```diff
@@ -0,0 +1,716 @@
+            )
+
+if($Online){
+        Set-Disk -Number ($Disk.Number) -IsOffline $false
+}
+elseif($Offline){
+        Set-Disk -Number ($Disk.Number) -IsOffline $true
+}
+
+}
+
+function Set-DiskWriteable{
+
+param(
+            $Disk
+            )
+
+Set-Disk -Number ($Disk.Number) -IsReadonly $false
+
+}
+
+function Search-Volume{
+
+param(
+            $Partition
+            )
+
+Get-Volume | Where-Object{$Partition.AccessPaths -like $_.ObjectId}      
+
+}
+
+function Set-Initialized{
+
+param(
+            $Disk,
+            $PartitionStyle
+            )
+
+$Disk| Initialize-Disk -PartitionStyle $PartitionStyle -Confirm:$false -PassThru
+}
+
+function Convert-PartitionStyle{
+
+param(
+            $Disk,
+            $PartitionStyle
+            )
+
+Invoke-Expression "'Select Disk $($Disk.Number)','Convert $($PartitionStyle)' | diskpart"
+}
+
+function Manage-ShellHWService{
+
+param(
+            $action
+            )
+
+switch($action){
+Stop {Stop-Service -Name ShellHWDetection -PassThru -ErrorAction Stop}
+Start {Start-Service -Name ShellHWDetection -PassThru -ErrorAction Stop}
+Check {(Get-Service ShellHWDetection -ErrorAction Stop).Status -eq "Running"}
+default {throw "Neither '-Stop', '-Start' nor 'Check' switch was passed to the function when invoked. Without the switches the service can not be maintained."}
+}
+}
+
+function Create-Partition{
+
+param(
+            $Disk,
+            $SetDriveLetter
+            )
+
+$Disk | New-Partition -UseMaximumSize -DriveLetter $SetDriveLetter
+
+}
+
+function Create-Volume{
+
+param(
+            $Volume,
+            $FileSystem,
+            $FileSystemLabel,
+            $FileSystemAllocUnitSize,
+            $FileSystemLargeFRS,
+            $FileSystemShortNames,
+            $FileSystemIntegrityStreams
+)
+
+$Alloc = $FileSystemAllocUnitSize *1KB
+
+[hashtable]$ParaVol = @{
+                                            FileSystem = $FileSystem
+                                            NewFileSystemLabel = $FileSystemLabel
+                                            AllocationUnitSize = $Alloc
+}
+
+if($FileSystem -eq "ntfs"){
+
+        if(!($FileSystemLargeFRS)){
+                $Volume | Format-Volume @ParaVol -ShortFileNameSupport $FileSystemShortNames -Force -Confirm:$false
+        }
+        else{
+                $Volume | Format-Volume @ParaVol -UseLargeFRS -ShortFileNameSupport $FileSystemShortNames -Force -Confirm:$false
+        }
+
+}
+elseif($FileSystem -eq "refs"){
+$Volume | Format-Volume @ParaVol -SetIntegrityStreams $FileSystemIntegrityStreams -Force -Confirm:$false
+}
+
+}
+
+$params = Parse-Args $args;
+
+# Create a new result object
+$result = [pscustomobject]@{changed=$false;log=[pscustomobject]@{};general_log=[pscustomobject]@{};parameters=[pscustomobject]@{};switches=[pscustomobject]@{}}
+
+## Extract each attributes into a variable
+# Find attributes
+$Size = Get-Attr -obj $params -name "disk_size" -resultobj $result -failifempty $true
```

Comment:
Use `Get-AnsibleParam -obj $params -name "disk_size" -type "str/int/bool/list/path" -failifempty $true` instead of `Get-Attr`. Also you don't need to pass in `-resultobj $result` as it really does nothing.